### DISCUSSION TITLE
Shall we allow multiple revealabes with same key?

### DISCUSSION THREAD

**Discussion by jayesh408** [svenjacobs/reveal#35] [Q&A] [answered]
We have a requirement where I want to highlight two components and have one explainer to explain them as their function remains same. 

E.g

There's a slider and at start & end of the slider there're `go left` `go right` button. I want to highlight both of them and have one explainer like what they do. 

as of now as I can you're having a `Set` to keep revealable in memory. Shall we have allow multiple revealbes with same key?

**Comment by svenjacobs** [comment]
This is an interesting use case 🤔 The problem is that an explanatory item which is specified via `overlayContent` is arranged according to the position and size of the revealable element. How should this work when there are multiple elements with the same ID? Where should the overlay content be placed? This would make the placement of the overlay content extremely difficult, maybe even impossible?

However what you can do is place both composables that you want to reveal at once in the same container, a `Box` for example. Then apply the `revealable()` modifier to the `Box`.

**Reply by jayesh408** [reply]
Yeh I also had thought around the same problem but then 

Maybe we can introduce something like `Revelable Group` and which allows adding including multiple revealables. and at the time of calling revealState.reveal(revealGroupId) and say align revelable
```
                                Modifier.align(
                                    verticalArrangement = RevealOverlayArrangement.Top,
                                    horizontalAlignment = Alignment.End,
                                    confineWidth = false,
                                    reveableKey = id_of_revealable_inside_group
                                )
```

**Reply by svenjacobs** [reply]
Why wouldn't a `Box` or another container work? Isn't that basically a "revealable group"?

**Reply by jayesh408** [reply]
Actually I'm having a row in which left and right icons are situated at the both edge of the screen and in between there's slider. 
I know its very specific usecase but Yeh sometimes you may wanna highlight components at top and bottom maybe

**Reply by svenjacobs** [reply, marked as answer]
I'm sorry, but I currently don't have a solution for you. Maybe you could use two separate revealables for both elements, one with "click here to go left" and one with "click here to go right". Would that be okay for you?

**Reply by jayesh408** [reply]
Wasn't that big problem! Thank you
